src/api/server.rs

```diff
@@ -140,16 +140,24 @@ fn make_routes(
     // Handle GraphQL queries. Headers will first be parsed to determine whether the query is
     // a subscription and if so, an attempt will be made to upgrade the connection to WebSockets.
     // All other queries will fall back to the default HTTP handler.
-    let graphql_handler = warp::path("graphql").and(graphql_subscription_handler.or(
-        async_graphql_warp::graphql(schema::build_schema().finish()).and_then(
-            |(schema, request): (Schema<_, _, _>, Request)| async move {
-                Ok::<_, Infallible>(GraphQLResponse::from(schema.execute(request).await))
-            },
-        ),
-    ));
+    let graphql_handler = if api.graphql {
+        warp::path("graphql")
+            .and(graphql_subscription_handler.or(
+                async_graphql_warp::graphql(schema::build_schema().finish()).and_then(
+                    |(schema, request): (Schema<_, _, _>, Request)| async move {
+                        Ok::<_, Infallible>(GraphQLResponse::from(schema.execute(request).await))
+                    },
+                ),
+            ))
+            .boxed()
+    } else {
+        warp::any()
+            .and_then(|| async { Err(warp::reject::not_found()) })
+            .boxed()
```

Comment:
This can be shortened to `not_found.boxed()` as below.

Reply:
When I replace that with `not_found.boxed()`, I encounter issues with the boxed instance of `not_found` that you are referring to below. For some reason, when I box `not_found` in this `else` statement, the second boxed `not_found` instance in the `else` statement below throws compilation error:

`if` and `else` have incompatible types
expected struct `BoxedFilter<(std::result::Result<http::Response<std::string::String>, http::Error>,)>`
   found struct `BoxedFilter<(warp::generic::Either<(warp::reply::WithHeader<impl Reply>,), (GraphQLResponse,)>,)>`
   
  I have tried playing around with cloning but to no avail, and for the sake of time I've gone with this approach. If you could point me in the right direction regarding how to fix this please let me know!

Reply:
The root problem here is that `not_found` takes on different inferred types in the two different usages that are incompatible. To resolve this without battling the type system, I suggest continuing with the above approach. Happy to defer to @bruceg if he knows of a more elegant approach.

As for the code structure, I recommend defining two variables so that the not found logic is consolidated in a single place.

```
        let not_found_graphql = warp::any().and_then(|| async { Err(warp::reject::not_found()) });
        let not_found_playground = warp::any().and_then(|| async { Err(warp::reject::not_found()) });
```

Reply:
Ah that makes sense... any opinions @bruceg?